project/npda/general_functions/ip_address.py

```diff
@@ -1,4 +1,6 @@
 def get_client_ip(request):
+    if request is None:
+        return "127.0.0.1"  # Default for tests/signals without request
```

Comment:
I think this would confusing in production logs - can we just leave it as `None`?